stdlib/buffer.mli

```diff
@@ -131,3 +131,8 @@ val add_channel : t -> in_channel -> int -> unit
 val output_buffer : out_channel -> t -> unit
 (** [output_buffer oc b] writes the current contents of buffer [b]
    on the output channel [oc]. *)
+
+val truncate : t -> int -> unit
+(** [truncate b len] truncates the length of [b] to [len]
+  Note: the intenal byte sequence is not shortenend.
+  Raise [Invalid_argument] if [len < 0] or [len > b.position]. *)
```

Comment:
`b.position` is an implementation detail, not exposed in the interface.  Better:
> `or [len > length b]`